### PR TITLE
Update architecture overview

### PR DESCRIPTION
**Pull request by jklukas**
This corrects info on how AET support is deployed, updates some outdated info
(sinks are now all Kube jobs rather than Dataflow), and now mentions the
concept of pipeline families.

### CODE REVIEW

**Comment by jklukas** [comment]
In particular, see the [updated diagram](https://raw.githubusercontent.com/mozilla/gcp-ingestion/7534f334733111246bf1661f10c194eb0e332bb5/docs/architecture/diagram.svg)

File: docs/architecture/overview.md [hunk 1]
```diff
@@ -7,21 +7,26 @@ This document specifies the architecture for GCP Ingestion as a whole.
...
+  only the "structured" pipeline family, but there are also deployments
+  for "telemetry", "stub-installer", and "pioneer"
+- The `Raw Sink` job copies messages from a PubSub `Raw Topic` to
   `BigQuery`
-- The Dataflow `Decoder` job decodes messages from PubSub `Raw Topics` to
-  PubSub `Decoded Topics`
-    - The Dataflow `Decoder` job checks for existence of `document_id`s in
+- The Dataflow `Decoder` job decodes messages from the PubSub `Raw Topic` to
+  the PubSub `Decoded Topic`
+    - Also checks for existence of `document_id`s in
```
Comment (whd) [inline docs/architecture/overview.md:18]: We've got a bunch of optional pipeline family features these days, such as debug topics, aet topics, and deduplication (which was originally not optional). Structured is technically not using dedup anymore, though having it in the diagram is illustrative.

File: docs/architecture/overview.md [hunk 2]
```diff
@@ -73,15 +78,21 @@ This document specifies the architecture for GCP Ingestion as a whole.
...
-    - The Pioneer project and Account Ecosystem Telemetry (AET) each have specific
-      requirements for decrypting values within the pipeline
-    - Private keys must be provided to the Decoder in encrypted form, to be decrypted
-      via Cloud KMS calls at startup and held only in memory
-    - Must remove or redact all AET `ecosystem_anon_id` values from the payload before
-      passing to any durable output, including errors
-    - Must have access restricted to a limited set of operators to avoid exposing private keys
-    - Encrypted fields must be JOSE JWE objects in Compact Serialization form
+
+### AET Decoder
```
Comment (whd) [inline docs/architecture/overview.md:82]: /CC @acmiyaguchi it might be good to add a section like this for pioneer.

**Comment by jklukas** [comment]
> LGTM except for maybe adding errors topics and sinks.

Added error topic and sink to the diagram.

**Pull request merged**